app/core/middleware.py

```
@@ -2,52 +2,152 @@
+        if hasattr(response, 'body'):
+            try:
+                original_body = response.body
+                if len(original_body) >= self.minimum_size:
+                    compressed_body = gzip.compress(original_body)
+
+                    # Only use compression if it actually reduces size
+                    if len(compressed_body) < len(original_body):
+                        response.headers["content-encoding"] = "gzip"
+                        response.headers["content-length"] = str(len(compressed_body))
+                        response.body = compressed_body
+            except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)